flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStore.java

```diff
@@ -258,16 +269,17 @@ private long discoveryIntervalMills() {
 
         private FileStoreSource buildFileSource(
                 boolean isContinuous, boolean continuousScanLatest) {
+            FileStore fileStore = buildFileStore();
             return new FileStoreSource(
-                    buildFileStore(),
+                    fileStore,
                     schema.primaryKeys().isEmpty(),
                     isContinuous,
                     discoveryIntervalMills(),
                     continuousScanLatest,
                     projectedFields,
                     partitionPredicate,
                     fieldPredicate,
-                    null);
+                    ((FileStoreImpl) fileStore).getSpecifiedPartitionedMeta());
```

Comment:
It makes more sense to get this `partitionedMeta` by parsing the flink options again (or adding an extra `withPartitionedMeta` in this `SourceBuilder` with the parsed value ) , rather than getting it from a casted `FileStore` implementation.

Reply:
> It makes more sense to get this `partitionedMeta` by parsing the flink options again (or adding an extra `withPartitionedMeta` in this `SourceBuilder` with the parsed value ) , rather than getting it from a casted `FileStore` implementation.

Actually, I did this in the previous version and wondered why we repeatedly parsed this value.